packages/amesos2/src/Amesos2_MultiVecAdapter_decl.hpp

```diff
@@ -284,24 +284,60 @@ namespace Amesos2 {
               const size_t ldx);
     };
 
+    // These methods take an input bInitialize and return a bool for bAssigned.
+    // The purpose of these two flags is to properly handle differences in logic
+    // for x and b when solving and avoid unecessary copies.
+    // Solvers should set bInitialize false for x and set bInitialize true for b.
+    // The view kokkos_vals can be uninitialized (0 size) when passed in.
+    // If the adapter can directly assign x or b to it's own memory spacee it will,
+    // and then no deep_copy will occur. Then bInitialize won't matter.
+    // However if the memory space or types don't match, the adapter will allocate
+    // the size of kokkos_vals and then, if bInitialize is true, copy it's values
+    // with deep_copy. This copy is only necessary for b, not x.
+    // If the adapter could directly assign the memory space the method will
+    // return true. Then the solver can skip doing a 'put X' because the solve
+    // happens directly in the adapter's memory space. Also Klu2 and SuperLU
+    // have special considerations because under some circumanstances they
+    // may solve b directly into it's own space. In that case, it's necessary
+    // to deep_copy b first or the adapter's b values will be permanently
+    // changed and the next solve cycle will fail. However if the method returns
+    // false then the adapter already had to deep_copy b so those solvers don't
+    // need to take any extra steps to protect the b memory space.
+    // Note, searching for bDidAssignX and bDidAssignB will find all places
```

Comment:
This explanation is helpful and important, but because of the length it may take a couple passes to find specific info, plus it requires hopping between solvers to get some clarification on usage and meaning. Can some of the comments here be modified, or copied and formatted differently after the explanation, to separate out key words/variables with summary of meaning of the value, something like:

```
do_get
  Return type (bool)
    bDidAssignX, bDidAssignB (bool): assignment state for solver
     true means ...
     false means ... 

Input
  bInitialize (bool): indicates in adapter if deep_copy occurred (it's not clear if a false means the deep_copy hasn't occurred and is still needed, or if false means a deep_copy is not necessary, can you clarify?)
    true means ...
    false means ...
```

My formatting isn't great but hopefully indicates what I think may be helpful for separating out the variables from the explanation with concise summaries that are easy to find (something similar to Tpetra's style for documenting functions and in/out variables may be better pattern to follow, doxygen is not necessary here)

Reply:
I have rewritten this to be more palatable. I left out the specific stuff on Klu2 and SuperLU. It's probably better just to leave those parts explained in the specific solvers which have their own comments.. Also removed the parts about the internal temporary copies since from the caller's point of view that doesn't matter for anything.